activerecord/lib/active_record/normalization.rb

```diff
@@ -82,7 +82,59 @@ def normalizes(*names, with:, apply_to_nil: false)
           NormalizedValueType.new(cast_type: cast_type, normalizer: with, normalize_nil: apply_to_nil)
         end
 
-        self.normalized_attributes += names.map(&:to_sym)
+        self.normalized_attributes += names.map { |name| resolve_attribute_name(name) }
+      end
+
+      # Similar to #normalizes, but normalizes attributes of the specified types.
+      #
+      #   class Snippet < ActiveRecord::Base
+      #     normalizes_attributes_of_type :string, :text, except: [:code], with: -> { _1.strip }
+      #   end
+      #
+      #   snippet = Snippet.create(title: "  Title", description: "Description.\n", code: "  code\n")
+      #   snippet.title        # => "Title"
+      #   snippet.description  # => "Description."
+      #   snippet.code         # => "  code\n"
+      #
+      # The normalization is applied to matching attributes from the model
+      # schema plus any attributes declared via
+      # {attribute}[rdoc-ref:ActiveRecord::Attributes::ClassMethods#attribute]
+      # _prior to_ the call to +normalizes_attributes_of_type+. Attributes
+      # declared after the call to +normalizes_attributes_of_type+ will not be
+      # affected. This behavior can be used to override a normalization from a
+      # parent class by redeclaring an attribute:
+      #
+      #   class ApplicationRecord < ActiveRecord::Base
+      #     normalizes_attributes_of_type :string, :text, with: -> { _1.strip }
+      #   end
+      #
+      #   class Snippet < ApplicationRecord
+      #     attribute :code, :text
+      #   end
+      #
+      #   snippet = Snippet.create(title: "  Title", description: "Description.\n", code: "  code\n")
+      #   snippet.title        # => "Title"
+      #   snippet.description  # => "Description."
+      #   snippet.code         # => "  code\n"
+      #
+      # ==== Options
+      #
+      # * +:except+ - Attributes to exclude from normalization. Can be a single
+      #   name or an array of names.
+      # * +:with+ - Any callable object that accepts the attribute's value as
+      #   its sole argument, and returns it normalized.
+      # * +:apply_to_nil+ - Whether to apply the normalization to +nil+ values.
+      #   Defaults to +false+.
+      #
+      def normalizes_attributes_of_type(*types, except: nil, with:, apply_to_nil: false)
```

Comment:
Although my first choice is two orthogonal, composable APIs like `normalizes` and `attributes_of_type`, I feel like rolling both into `normalizes` isn't quite right.

Visually, it could be similar:

```ruby
normalizes_attributes_of_type :string, with: -> { _1.strip }
normalizes attributes_of_type: :string, with: -> { _1.strip }
```

But with the latter, we have to handle the case where someone specifies both names and types.  Should names + types denote union or intersection, or should it raise an error?  My instinct would be to raise an error.  In which case, semantically, we _are_ defining two APIs, so why not have the code reflect that?

Rolling both into a single API would be like combining `has_many` and `has_and_belongs_to_many` into a single API with a `:belongs_to_many` kwarg.

Reply:
Small suggestion:
To stay consistent with `normalizes` (which does not include attribute as well) I'd drop the `attribute` part of the method name (or kwarg).

```ruby
normalizes_type :string, with: -> { _1.strip }
normalizes type: :string, with: -> { _1.strip }

vs.

normalizes_attributes_of_type :string
normalizes attribute_of_type: :string
```

Reply:
> What about normalizes_each? 

Better IMO, but as I said I don't really feel that strongly about it. I'd appreciate someone else input here.

Thanks for removing the `except / only`.